src/Containers/Microsoft.NET.Build.Containers/ContainerHelpers.cs

```diff
@@ -143,13 +144,39 @@ internal static bool IsValidImageTag(string imageTag)
         return ReferenceParser.anchoredTagRegexp.IsMatch(imageTag);
     }
 
+
     /// <summary>
     /// Given an already-validated registry domain, this is our hueristic to determine what HTTP protocol should be used to interact with it.
+    /// If the domain is localhost, we default to HTTP. Otherwise, we check the Docker config to see if the registry is marked as insecure.
     /// This is primarily for testing - in the real world almost all usage should be through HTTPS!
     /// </summary>
     internal static Uri TryExpandRegistryToUri(string alreadyValidatedDomain)
     {
-        var prefix = alreadyValidatedDomain.StartsWith("localhost", StringComparison.Ordinal) ? "http" : "https";
+        string prefix = "https";
+        if (alreadyValidatedDomain.StartsWith("localhost", StringComparison.Ordinal))
+        {
+            prefix = "http";
+        }
+#if !NET472
+        //check the docker config to see if the registry is marked as insecure
+        if (DockerCli.GetDockerConfig().RootElement.TryGetProperty("RegistryConfig", out var registryConfig) && registryConfig.ValueKind == System.Text.Json.JsonValueKind.Object)
+        {
+            if(registryConfig.TryGetProperty("IndexConfigs",out var indexConfigs) && indexConfigs.ValueKind == System.Text.Json.JsonValueKind.Object){
+                foreach (var property in indexConfigs.EnumerateObject())
+                {
+                    if(property.Value.ValueKind == System.Text.Json.JsonValueKind.Object && property.Value.TryGetProperty("Secure",out var secure) && !secure.GetBoolean())
+                    {
+                        if (property.Name.Equals(alreadyValidatedDomain,StringComparison.Ordinal))
+                        {
+                            prefix = "http";
+                            break;
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
I think this should be a dedicated method in the `DockerCli` class - something like `bool DockerCli.IsInsecureRegistry(string registryDomain)`.

The `DockerCli` type has to understand how both Docker and Podman work - do you know anything about how Podman manages this concept of an insecure registry?

Reply:
For podman, `podman info --format="{{json .}}` will have a structure containing the known registry information:

```json
... some prior content
"registries": {
    "localhost:32000": {
      "Prefix": "localhost:32000",
      "Location": "localhost:32000",
      "Insecure": true,
      "PullFromMirror": "",
      "Mirrors": null,
      "Blocked": false,
      "MirrorByDigestOnly": false
    }
},
... some other content
```

So we should be able to use the `registries` object in much the same way as the Docker object you've already found, just checking the value of the `Insecure` property.